src/web/components/ApiKeyManagement/KeyEditDialog.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ApiKeyDTO } from '../../../api/services/adminServiceHelpers';
+import { EditApiKeyFormDTO } from '../../services/apiKeyService';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { TextInput } from '../Input/TextInput';
+
+export type OnApiKeyEdit = (
+  form: EditApiKeyFormDTO,
+  setApiKey: React.Dispatch<React.SetStateAction<ApiKeyDTO>>,
+  participantId?: number
+) => void;
+
+type KeyEditDialogProps = {
+  onEdit: OnApiKeyEdit;
+  triggerButton: JSX.Element;
+  availableRoles: ApiRoleDTO[];
+  apiKey: ApiKeyDTO;
+  setApiKey: React.Dispatch<React.SetStateAction<ApiKeyDTO>>;
+};
+
+function KeyEditDialog({
+  availableRoles,
+  onEdit,
+  triggerButton,
+  apiKey,
+  setApiKey,
+}: KeyEditDialogProps) {
+  const [open, setOpen] = useState(false);
+
+  const onSubmit: SubmitHandler<EditApiKeyFormDTO> = async (formData) => {
+    await onEdit(formData, setApiKey);
+    setOpen(false);
+  };
+
+  const possibleRolesMap = new Map<number, ApiRoleDTO>();
+  availableRoles.map((role) => possibleRolesMap.set(role.id, role));
+  apiKey.roles.map((role) => possibleRolesMap.set(role.id, role));
+
+  const defaultFormData: EditApiKeyFormDTO = {
+    keyId: apiKey.key_id,
+    newName: apiKey.name,
+    newApiRoles: apiKey.roles.map((role) => role.roleName),
+  };
+
+  return (
+    <Dialog
+      closeButtonText='Cancel'
+      open={open}
+      onOpenChange={setOpen}
+      triggerButton={triggerButton}
+      title={`Edit ${apiKey.name}`}
+    >
+      <Form<EditApiKeyFormDTO>
+        onSubmit={onSubmit}
+        defaultValues={defaultFormData}
+        submitButtonText='Edit'
```

Comment:
I'm thinking this should be `'Save Key'` or similar, for consistency
![Uploading image.png…]()